src/dist/dist.rs

```diff
@@ -172,43 +190,25 @@ impl FromStr for ParsedToolchainDesc {
             static ref TOOLCHAIN_CHANNEL_RE: Regex = Regex::new(&TOOLCHAIN_CHANNEL_PATTERN).unwrap();
         }
 
-        let d = TOOLCHAIN_CHANNEL_RE.captures(desc).map(|c| {
-            fn fn_map(s: &str) -> Option<String> {
-                if s.is_empty() {
-                    None
-                } else {
-                    Some(s.to_owned())
-                }
-            }
+        TOOLCHAIN_CHANNEL_RE
```

Comment:
I think this way of writing this would be clearer:

```rust
let channel = match TOOLCHAIN_CHANNEL_RE.captures(desc) {
    Some(c) => c,
    None => return RustupError::InvalidToolchainName(desc.to_owned().into()),
};

..
```